jaxrs/src/main/java/io/airlift/jaxrs/AsyncResponseBinder.java

```diff
@@ -0,0 +1,130 @@
+package io.airlift.jaxrs;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.Duration;
+import jakarta.servlet.AsyncEvent;
+import jakarta.servlet.AsyncListener;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.ws.rs.container.AsyncResponse;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Response;
+
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static io.airlift.jaxrs.AsyncResponseHandler.bindAsyncResponse;
+import static jakarta.ws.rs.core.Response.status;
+import static java.util.Objects.requireNonNull;
+
+public class AsyncResponseBinder
+{
+    private final HttpServletRequest servletRequest;
+    private boolean cancelOnClientDisconnect = true;
+    private ListenableFuture<?> future;
+    private Executor httpExecutor;
```

Comment:
I don't like that these required fields need to be set using the builder methods. I think it would be better to make them required arguments to `bindTo()` similar to `bindAsyncResponse()`.

Reply:
Makes sense. Moved to method